### DISCUSSION TITLE
extract32/64 installer options

### DISCUSSION THREAD

**Discussion by The-Lone-Funman** [Open-Shell/Open-Shell-Menu#1278] [Ideas]
Love the updates and fixes in the pre-releases, miss the msi installer extract32 extract64 installation options, which helps greatly in creating portable installations -- and hope these options are added back in the future. Thanks for all the hard (and tedious) work keeping Open-Shell-Menu current!

**Comment by ge0rdi** [comment]
Hmm, `extract32`/`extract64` command line options work for me as usual.
Running `OpenShellSetup_4_4_187.exe extract64` does extract 64bit MSI.

**Comment by The-Lone-Funman** [comment]
Thankyou for your reply! Yes, I just re-downloaded it and it does work! I got the previous 4.4.187 download from MajorGeeks (hadn't known there had been any updates since 4.4.170 till I saw it there), and extract32 and extract64 not only didn't work, they weren't even in the command line switches pop-up help...

**Comment by ge0rdi** [comment]
> I got the previous 4.4.187 download from MajorGeeks

Please, always download from original source, which is Github in case of `Open-Shell`.
We cannot guarantee that other sites provide our files in exactly the same way as we do.

> they weren't even in the command line switches pop-up help

They are not listed and never were.

**Reply by The-Lone-Funman** [reply]
> Please, always download from original source, which is Github in case of `Open-Shell`. We cannot guarantee that other sites provide our files in exactly the same way as we do.
> 
Yes it's not something I normally do, it was just the version update that I hadn't seen here as the pre-release versions aren't listed on the main page, and I thought someone had perhaps picked it up and created a fork to add something, and wanted to investigate...

> They are not listed and never were.
>
They sure appear to be for me, in all the versions I've downloaded here including 4.4.170 and 4.4.187 (launch the installer with parameter /? and see for yourself):

![image](https://user-images.githubusercontent.com/118757032/211188767-8622984e-2215-4db0-ba72-0245c776ce1a.png)

I hope there's no intent to remove this, there were quite a few people that were pleased when it was added and/or documented on the original Classic Shell site and forums...

**Reply by ge0rdi** [reply]
> They sure appear to be for me, in all the versions

Ah, sorry. I must have mixed it with something else I guess.

> I hope there's no intent to remove this

There are no such plans AFAIK.

**Reply by The-Lone-Funman** [reply]
I think the confusion, and I've done this too, is if you launch with /h or /? you get a different set of parameters. Thanks for replying!

**Reply by ge0rdi** [reply]
Right.
`Open-Shell` setup recognizes `help` and `/?` parameters and then displays its help window.
`/h` is not recognized and thus passed to `msiexec` that displays its help window.

I guess it would make sense to add `/h` and `/help` command line parameters to our setup to prevent such confusion in the future.
